hugolib/site.go

```diff
@@ -99,7 +99,7 @@ type SiteInfo struct {
 	Social                SiteSocial
 	Sections              Taxonomy
 	Pages                 *Pages
-	Files                 []*source.File
+	Files                 *[]*source.File
```

Comment:
If I grok go right, in both cases (Pages and Files), the information is stored in `Site`. `SiteInfo` contains a pointer to the arrays in `Site`. Until this PR, `SiteInfo.Files` wasn't being used or initialized - now it is.